src/com/aca/exceptions/WrongProductData.java

```diff
@@ -0,0 +1,6 @@
+package com.aca.exceptions;
+
+import java.io.IOException;
+
+public class WrongProductData extends IOException {
```

Comment:
IOException shows that you have technical problems in I/O operations, but you don't have it here, everything is ok with input output, just you don't accept empty files which is business logic, so this should be some RuntimeException not IO